providers/unleash/pkg/provider.go

```diff
@@ -0,0 +1,236 @@
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.IntResolutionDetail{
+		Value:                    res.Value.(int64),
+		ProviderResolutionDetail: res.ProviderResolutionDetail,
+	}
+}
+
+func (p *Provider) StringEvaluation(ctx context.Context, flag string, defaultValue string, evalCtx of.FlattenedContext) of.StringResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.StringResolutionDetail{
+		Value: fmt.Sprint(res.Value),
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			ResolutionError: res.ProviderResolutionDetail.ResolutionError,
+			Reason:          res.ProviderResolutionDetail.Reason,
+			Variant:         res.Variant,
+			FlagMetadata:    res.FlagMetadata,
+		},
+	}
+}
+
+func (p *Provider) ObjectEvaluation(ctx context.Context, flag string, defaultValue interface{}, evalCtx of.FlattenedContext) of.InterfaceResolutionDetail {
+	if p.status != of.ReadyState {
+		if p.status == of.NotReadyState {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewProviderNotReadyResolutionError(providerNotReady),
+					Reason:          of.ErrorReason,
+				},
+			}
+		} else {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	unleashContext, err := toUnleashContext(evalCtx)
+	if err != nil {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError(err.Error()),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	variant := unleash.GetVariant(flag, unleash.WithVariantContext(*unleashContext))
+	flagMetadata := map[string]interface{}{
+		"enabled": variant.Enabled,
+	}
+	if variant.Name == api.DISABLED_VARIANT.Name {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				Variant:      "",
+				FlagMetadata: flagMetadata,
+			},
+		}
+	} else {
+		return of.InterfaceResolutionDetail{
+			Value: variant.Payload.Value,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				Variant:      variant.Name,
+				FlagMetadata: flagMetadata,
+			},
+		}
+	}
+}
+
+func toUnleashContext(evalCtx of.FlattenedContext) (*unleashContext.Context, error) {
+	if len(evalCtx) == 0 {
+		return &unleashContext.Context{}, nil
+	}
+
+	unleashContext := &unleashContext.Context{}
+
+	custom := make(map[string]string)
+	for key, origVal := range evalCtx {
+		val, ok := toStr(origVal)
+		if !ok {
+			return nil, fmt.Errorf("key `%s` can not be converted to string", key)
```

Comment:
May want to use "Invalid Context" error here: https://github.com/open-feature/go-sdk/blob/d8da306f06a5159abdb088b91e78fb59cb0b68a5/pkg/openfeature/resolution_error.go#L19